R/guide-colorbar.r

```diff
@@ -344,17 +349,48 @@ guide_gengrob.colorbar <- function(guide, theme) {
   hgap <- width_cm(theme$legend.spacing.x  %||% (0.5 * unit(title_fontsize, "pt")))
   vgap <- height_cm(theme$legend.spacing.y %||% (0.5 * unit(title_fontsize, "pt")))
 
-  # label
+  # Labels
+
+  # get the defaults for label justification. The defaults are complicated and depend
+  # on the direction of the legend and on label placement
+  just_defaults <- label_just_defaults.colorbar(guide$direction, label.position)
+  # don't set expressions left-justified
+  if (just_defaults$hjust == 0 && any(is.expression(guide$key$.label))) just_defaults$hjust <- 1
+
+  # get the label theme
   label.theme <- guide$label.theme %||% calc_element("legend.text", theme)
+
+  # We break inheritance for hjust and vjust, because that's more intuitive here; it still allows manual
+  # setting of hjust and vjust if desired. The alternative is to ignore hjust and vjust altogether, which
+  # seems worse
+  if (is.null(guide$label.theme$hjust) && is.null(theme$legend.text$hjust)) label.theme$hjust <- NULL
+  if (is.null(guide$label.theme$vjust) && is.null(theme$legend.text$vjust)) label.theme$vjust <- NULL
+
+  # label.theme in param of guide_legend() > theme$legend.text.align > default
+  hjust <- guide$label.hjust %||% theme$legend.text.align %||% label.theme$hjust %||%
+    just_defaults$hjust
+  vjust <- guide$label.vjust %||% label.theme$vjust %||%
+    just_defaults$vjust
+
   grob.label <- {
     if (!guide$label)
       zeroGrob()
     else {
-      hjust <- x <- guide$label.hjust %||% theme$legend.text.align %||%
-        if (any(is.expression(guide$key$.label))) 1 else switch(guide$direction, horizontal = 0.5, vertical = 0)
-      vjust <- y <- guide$label.vjust %||% 0.5
-      switch(guide$direction, horizontal = {x <- label_pos; y <- vjust}, "vertical" = {x <- hjust; y <- label_pos})
-
+      switch(
```

Comment:
Since you're rewriting this, do you mind moving the assignment into the individual blocks of the if statement? I think that will make it easier to read.

Reply:
Just so I understand what you're requesting. Patterns such as:
```{r}
switch(
  guide$direction,
  "horizontal" = {
    ...
  },
  "vertical" = {
  }
)
```
are used all over the place in this code. Do you want me to replace those with the following?
```{r}
if (guide$direction == "horizontal") {
  ...
}
else {
  ...
}
```

Reply:
Yes please!

Reply:
I had originally not understood the "move assignment" part of the request, but I think I got it now. The code is easier to read now.